pkgs/servers/code-server/default.nix

```
@@ -1,26 +1,26 @@
 { lib, stdenv, fetchFromGitHub, buildGoModule, makeWrapper, runCommand
-, moreutils, jq, git, zip, rsync, pkg-config, yarn, python2
-, nodejs-12_x, libsecret, xorg, ripgrep
+, moreutils, jq, git, zip, rsync, pkg-config, yarn, python3
+, nodejs-14_x, libsecret, xorg, ripgrep
 , AppKit, Cocoa, Security, cctools }:
 
 let
   system = stdenv.hostPlatform.system;
 
-  nodejs = nodejs-12_x;
-  python = python2;
+  nodejs = nodejs-14_x;
+  python = python3;
   yarn' = yarn.override { inherit nodejs; };
   defaultYarnOpts = [ "frozen-lockfile" "non-interactive" "no-progress"];
 
 in stdenv.mkDerivation rec {
   pname = "code-server";
-  version = "3.8.0";
-  commit = "c4610f7829701aadb045d450013b84491c30580d";
+  version = "3.9.0";
+  commit = "fc6d123da59a4e5a675ac8e080f66e032ba01a1b";
 
   src = fetchFromGitHub {
     owner = "cdr";
     repo = "code-server";
-    rev = "v${version}";
-    sha256 = "1snc7dbqfz53337h6av2zhkrn54ypanxljs5by4jqczq96c2v6yk";
+    rev = commit;
```

```suggestion
    rev = "v${commit}";
```

This is safer to prevent fetching commits on forks accidentally.

---

For future upgrade of `code-server`, we may try to fetch a later commit that doesn't have functionality changes, only things like dependabot fixes so that the `yarn.lock` things are no longer blockers? So I'd argue that commits make lives easier for future upgraders.

---

I am not aware of any blockers when updating the yarn.lock but fetching a later commit would require to change the version number to unstable-XXXX-XX-XX.

---

Fair point, changed back to `v${version}`.